### PR TITLE
Detect NEXT_PUBLIC_WORDPRESS_URL using wpengine.com TLD and recommend using wpenginepowered.com TLD

### PR DESCRIPTION
**Pull request by blakewilson**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.
- [x] If a code change, I have written testing instructions that the whole team & outside contributors can understand.
- [x] I have written and included a comprehensive [changeset](https://github.com/wpengine/faustjs/blob/canary/DEVELOPMENT.md#deployment) to properly document the changes I've made.

## Description

This PR recommends the user update their NEXT_PUBLIC_WORDPRESS_URL to use the `wpenginepowered.com` TLD if they are using the `wpengine.com` TLD.

## Testing

Unit tests have been provided. However, you can manually test this PR by:

1. Checking out this branch
2. `npm install`
3. `npm run build`
4. Set the `NEXT_PUBLIC_WORDPRESS_URL` in `examples/next/faustwp-getting-started/.env.local` to a `https://xxxxx.wpengine.com` URL.
5. Run the example project `npm run dev -w @faustwp/getting-started-example`
6. Notice the info text in the terminal
    Screenshot 2024-02-14 at 11 52 49 AM

### CODE REVIEW

File: packages/faustwp-cli/tests/healthCheck/validateFaustEnvVars.test.ts [hunk 1]
```diff
@@ -50,3 +53,27 @@ describe('healthCheck/validateFaustEnvVars', () => {
...
+
+  it('does not match urls that are not subdomains of wpengine.com', () => {
+    expect(isWPEngineComSubdomain('https://example.com')).toBeFalsy();
+    expect(isWPEngineComSubdomain('https://wpengine.com')).toBeFalsy();
+    expect(isWPEngineComSubdomain('https://wpengine.com/plans')).toBeFalsy();
+    expect(isWPEngineComSubdomain('https://my-site.wpengine.co')).toBeFalsy();
+    expect(
+      isWPEngineComSubdomain('https://my-site.wpenginepowered.com'),
+    ).toBeFalsy();
+    expect(isWPEngineComSubdomain('https://my-site.wpengine.co'));
```
Comment (mindctrl) [inline packages/faustwp-cli/tests/healthCheck/validateFaustEnvVars.test.ts:77]: This looks like a partial dupe of line 73.

```suggestion
```

Comment (blakewilson) [reply packages/faustwp-cli/tests/healthCheck/validateFaustEnvVars.test.ts:77]: @mindctrl Good eye. thanks!

File: packages/faustwp-cli/src/healthCheck/validateFaustEnvVars.ts [hunk 2]
```diff
@@ -1,5 +1,15 @@
 import { getWpSecret } from '../utils/index.js';
-import { errorLog, warnLog } from '../stdout/index.js';
+import { errorLog, infoLog, warnLog } from '../stdout/index.js';
+
+export function isWPEngineComSubdomain(url: string) {
+  const regex = /\b\w+\.wpengine\.com\b/;
+
+  if (regex.test(url)) {
```
Comment (theodesp) [inline packages/faustwp-cli/src/healthCheck/validateFaustEnvVars.ts:7]:
```suggestion
 return regex.test(url);
```

Comment (theodesp) [reply packages/faustwp-cli/src/healthCheck/validateFaustEnvVars.ts:7]: Since regex.test(url) is true if it matches you can just return the value here.

Comment (blakewilson) [reply packages/faustwp-cli/src/healthCheck/validateFaustEnvVars.ts:7]: @theodesp Good call

**Review by mindctrl — APPROVED** [review]
Love it.

Example test:

```
info - Found NEXT_PUBLIC_WORDPRESS_URL using wpengine.com TLD.
info - It is recommended to use the wpenginepowered.com TLD instead.
info - Ex: https://example.wpengine.com -> https://example.wpenginepowered.com
info - This will leverage WP Engine's Advanced Network CDN. See: https://wpengine.com/support/network/
```

**Review by TeresaGobble — APPROVED** [review]
Nicely done Blake! Everything looks great to me. 

image

**Review by theodesp — APPROVED** [review]

**Pull request merged**